pkg/didmethod/peer/resolver.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package peer
+
+import (
+	errors "golang.org/x/xerrors"
+)
+
+// DIDResolver resolver
+type DIDResolver struct {
+	store *DIDStore
+}
+
+// NewDIDResolver new Peer DID resolver instance
+func NewDIDResolver(store *DIDStore) *DIDResolver {
+	return &DIDResolver{store: store}
+}
+
+// Read implements DidMethod.Read function
+func (resl *DIDResolver) Read(did string, versionID interface{}, versionTime string, noCache bool) ([]byte, error) {
+	// validate the input
+	if did == "" {
```

Comment:
Will store.Get fail in some strange way if did == ""?

Or produce a normal error that DID was not found. If so, I wonder why this check is needed?

Reply:
If we remove the validation, the error that we get is not easily readable.

Error description when we remove it.
```
Fetching data from store failed:
     github.com/hyperledger/aries-framework-go/pkg/didmethod/peer.(*DIDResolver).Read
         /Users/rolson.quadras/gows/src/github.com/hyperledger/aries-framework-go/pkg/didmethod/peer/resolver.go:33
   - ID is mandatory:
     github.com/hyperledger/aries-framework-go/pkg/didmethod/peer.(*DIDStore).Get
         /Users/rolson.quadras/gows/src/github.com/hyperledger/aries-framework-go/pkg/didmethod/peer/store.go:78
```

Reply:
I'd argue (IMO):
- This doesn't seem like a common scenario - why would an empty DID be provided anyways?
- we should minimize input checks that don't result in serious issues like a panic or causing corruption.
- error checking should be handled as close as possible to where the real check is needed, and bubble it back up (rather than duplicating at every level).
- parameters should be documented in the code (and, hence, GoDoc) comments: what is each argument for? Is it required? If not, how do I not include?
  - I generally think "just in case" checks for a human doing something very silly should be minimized (i.e., is this a check against a "silly" human error or logic error elsewhere in the code?).
-  Ideally zero-values are treated as defaults, when possible.

Reply:
agreed with most of the points. [Removed](https://github.com/hyperledger/aries-framework-go/compare/2edd9c351304abb26a71e1406933eea923d0c228..2e43c02af5c1920a233d76a7fb6f3430ef827bc6) the validation.